nvd/nvd.go

```diff
@@ -99,11 +100,16 @@ func (u Updater) Update() error {
 			}
 			slog.Info("Fetched NVD entries", slog.Int("total", totalResults), slog.Int("start_index", startIndex))
 		}
+		totalEntries += totalResults
 	}
 
-	// Update last_updated.json at the end.
-	if err = utils.SetLastUpdatedDate(apiDir, u.lastModEndDate); err != nil {
-		return xerrors.Errorf("unable to update last_updated.json file: %w", err)
+	// If NVD didn't return records for all intervals
+	// we shouldn't update the last update as it might be a bug.
+	if totalEntries > 0 {
+		// Update last_updated.json at the end.
+		if err = utils.SetLastUpdatedDate(apiDir, u.lastModEndDate); err != nil {
+			return xerrors.Errorf("unable to update last_updated.json file: %w", err)
+		}
```

Comment:
nit: I prefer the following style.

>Don’t hide happy path logic inside a nest of indented braces

https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88

```suggestion
	if totalEntries == 0 {
		return nil
	}
```

Reply:
fixed in https://github.com/aquasecurity/vuln-list-update/pull/328/commits/86d43fe7492455d9130475e2da8aa3a1f63bb3c4